File: loans-service/Events/AccountEvent.cs
namespace loans_service.Events;

public class AccountEventData
{
    public string Id { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public decimal Balance { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }
}

public class AccountEvent
{
    public const string ExchangeName = "lending.events";

    public string EventId { get; set; } = string.Empty;

    public string EventType { get; set; } = string.Empty;

    public DateTime OccurredAt { get; set; }

    public AccountEventData Data { get; set; } = new();
}

File: accounts-service/src/accounts/events/account.event.ts
export interface AccountEventPayload {
  id: string;
  name: string;
  email: string;
  /**
   * Decimal columns are read as decimal strings.
   */
  balance: string;
  createdAt: string;
  updatedAt: string;
}

export interface AccountEvent {
  eventId: string;
  eventType: 'account.created' | 'account.updated' | 'account.deleted';
  occurredAt: Date;
  correlationId?: string;
  data: Partial<AccountEventPayload>;
}

export const ACCOUNT_EVENTS_EXCHANGE = 'lending.events';
